### ISSUE TITLE
GPS Drifting Lately

### ISSUE THREAD

**Issue by broyuken** [gcobb321/icloud3#32]
Latest version of HA and iCloud3, all of a sudden my GPS has been wandering and marking me and my wife as not home when I am home. I believe this started right around the time that I updated the event log card from v1 to v2.x. It just switches to that from home for a few minutes and then back again.

07:01:45 | not_home | Away | 2.5 min |   | 0.06 mi
-- | -- | -- | -- | -- | --
  | Results:  BaseZone-home > Zone-not_home, GPS-(xxxxxxxxxxxxxxx, xxxxxxxxxxxxxxxxx),  Interval-2.5 min, Dist-0.06 mi, TravTime- (not_set), NextUpdt-7:04:15,  Located-7:01:31 (13 sec ago)

-----------------------------------------------------------------

07:06:00 | home | Home | 2 hrs |   | 0 mi
-- | -- | -- | -- | -- | --
  | Moving to zone center > home, GPS-(42.162324, -71.540265) to (xxxxxxxxxxxxxxx, xxxxxxxxxxxxxx), Dist-36.42m
07:06:00 | home | Home | 2 hrs |   | 0 mi
  | Results:  BaseZone-home > Zone-home, GPS-(xxxxxxxxxxxx, xxxxxxxxxxxxxx), Interval-2  hrs, Dist-0 mi, TravTime- (in_zone), NextUpdt-9:06:00, Located-7:05:49  (11 sec ago)

**Comment by gcobb321** [comment]
I found a problem in the routines that check to see if the device had been moved out of the zone a short distance without a zone exit trigger. The distance test was using meters for the current distance (96m in your case) and kilometers for the MoveBackIntoZone distance (.240m in your case when it should have been 240m). 

The updated iCloud3 code is in the [development directory](https://github.com/gcobb321/icloud3/tree/master/development%20v2.1.0b-03.27.2020).

Download it and see if your problem still exists.

**Comment by broyuken** [comment]
I just downloaded it, sorry have been pretty busy the past few weeks and haven't had a chance to do this until now. I have also noticed that my phone will send me notifications saying my stationary zone is not loaded. Are those related by any chance?

**Comment by broyuken** [comment]
Nope, just had another drift after using the new dev build.

**Comment by gcobb321** [comment]
I have found that it was drifting and not being caught when the Next Update Time was reached but was being caught and discarded on a trigger from the iOS app.

 I have fixed that and also have made changes in the way locations with old location times are handled. One involves looking ahead for the next update time trigger and requesting a location update before it occurs. When the next update time is then reached, a good location is available instead of it being old. It is working for FmF tracking but I still need to get it running for FamShr tracking before putting up the next development release. Right now, I have things in a mess but should have everything working in a few days.

————————
Gary Cobb
